instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/__init__.py

```diff
@@ -248,8 +249,14 @@ def _trace_retry(*args, **kwargs):
         span.set_attribute(_TASK_RETRY_REASON_KEY, str(reason))
 
 
-def carrier_extractor(carrier, key):
-    value = getattr(carrier, key, [])
-    if isinstance(value, str) or not isinstance(value, Iterable):
-        value = (value,)
-    return value
+class Getter:
+    @staticmethod
+    def get(carrier, key):
+        value = getattr(carrier, key, [])
+        if isinstance(value, str) or not isinstance(value, Iterable):
+            value = (value,)
+        return value
+
+    @staticmethod
+    def keys(carrier):
+        return carrier.keys()
```

Comment:
This feels a bit too verbose just to satisfy an interface. Can we have a convenience function that creates one for us? Something like:

```python
class Getter:
    def __init__(self, get=default_get, keys=default_keys):
        self._get = get
        self._keys = keys

    def get(self, carrier, key):
         return self._get(carrier, key)

    def keys(self, carrier):
         return self._keys(carrier)
```

Then auto and manual instrumented code can use it like:

```
getter = Getter(get=carrier_extractor)
```

This should reduce some boilerplate and allow users to just pass an inline lambda function if that'll get the job done instead of writing the whole class with both methods.

Reply:
@owais Implemented this and added comments below